lib/matplotlib/contour.py

```diff
@@ -1027,6 +1027,15 @@ def _make_paths(self, segs, kinds):
             return [mpath.Path(seg) for seg in segs]
 
     def changed(self):
+        if not hasattr(self, "cvalues"):
```

Comment:
this behaviour seems new - can you clarify why you need this?  The explanation below didn't help me understand why this is needed now.

Reply:
This is actually to keep the behavior the same as before. I kept the commits that fix the tests in with messages to try and explain why I did this. (See the second and third commit messages)

At a high level though, this `changed()` overrides the one in ScalarMappable and we need to make sure that we are "changing" in the appropriate place. This was a little odd because contour does some extra things with cvalues and levels that aren't needed on any of the other mappables, so we need to autoscale and adjust the limits in the right order.

Reply:
The joys of making things a bit lazier!